dotcom-rendering/src/web/components/TableOfContents.importable.tsx

```diff
@@ -93,15 +84,12 @@ const chevronPosition = css`
 
 export const TableOfContents = ({ tableOfContents, format }: Props) => {
 	const palette = decidePalette(format);
-	const [open, setOpen] = useState(tableOfContents.length < 5);
+	const hasMoreThan5Items = tableOfContents.length < 5;
```

Comment:
I think there might be a naming issue here which threw me - do we mean `>`?
```ts
const hasMoreThan5Items = tableOfContents.length > 5;
```

Then `const [open, setOpen] = useState(hasMoreThan5Items);` and  `open={open}` would work as expected which feels more intuitive.

Reply:
We would need to have the table open by default if `tableOfContents.length < 5` and closed if `tableOfContents.length > 5`. This is just the initial state, after then user should control the state by clicking. But thanks to you, your question made me search more as I was confused by the problem 🤔 

Reading this issue https://github.com/facebook/react/issues/15486 I found that this problem happens when we want to control the state of the details with react. The suggested solution in the issue worked for us too: adding `event.preventDefault()` to the onClick handler.

More details about it can be read in this post: https://github.com/facebook/react/issues/15486#issuecomment-873516817 :

_Unlike most other native DOM elements, the details element has state which it can update by itself without React knowing about it._

_the solution is to call event.preventDefault() in the onClick handler to disable the browser's default behavior so that React and only React controls the state_

Reply:
This makes a tonne of sense - great investigation!